### ISSUE TITLE
Add categories API endpoint to just provide a list of categories, same for predicates, etc

### ISSUE THREAD

**Issue by kevinschaper** [monarch-initiative/monarch-app#159]

**Comment by vincerubinetti** [comment]
Is this referring to producing a list of all possible categories/predicates/etc?

If only needed for the frontend, a better way to do this would be to have this generated automatically in the `model.ts` file. Example:

```typescript
const categories = [
  "gene",
  "disease",
  "phenotype",
   ...
] as const;

type Category = (typeof categories)[number];

type SearchResults = {
  results: {
    title: string,
    category: Category,
    ...
  }[]
}
```

Now `Category`, instead of being a plain `string`, is `"gene" | "disease" | "phenotype"`. It now has to be one of those values instead of any kind of string, and that kind of type specificity could help the frontend typescript a lot.

If this is just to help users using the API directly know what types of values could be returned or used, perhaps it's possible to define the schema in such a way that the Swagger UI will show the possible values in the return type schema and input parameters, instead of just a string.

**Comment by kevinschaper** [comment]
@vincerubinetti Yup, using an enumeration would be ideal for the API also - since docs do populate with enumeration values as a select box. It looks like typescriptgen doesn't have enumeration support, so we're already missing out on the association type enum. I'll make an issue for that.

I worry a little that it can be more brittle to use an enumeration and it more tightly couples API versions to specific data releases. I think it's worth trying though. 

I think my approach is going to be to make the big lift of putting category and predicate enumerations into the monarch-py model, and then using them everywhere in monarch-ingest. Hopefully this means that we won't add values to the graph that aren't in the enum.
